packages/richtext-lexical/src/lexical/plugins/TextPlugin/index.tsx

```diff
@@ -0,0 +1,43 @@
+'use client'
+import type { TextFormatType } from 'lexical'
+
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext'
+import { TEXT_TYPE_TO_FORMAT, TextNode } from 'lexical'
+import { type SanitizedClientFeatures } from 'packages/richtext-lexical/src/features/typesClient.js'
+import { useEffect } from 'react'
+
+export function TextPlugin({ features }: { features: SanitizedClientFeatures }) {
+  const [editor] = useLexicalComposerContext()
+
+  useEffect(() => {
+    const disabledFormats = getDisabledFormats(features.enabledFormats as TextFormatType[])
+    if (disabledFormats.length === 0) {
+      return
+    }
+    // Ideally override the TextNode with our own TextNode (changing its setFormat or toggleFormat methods),
+    // would be more performant. If we find a noticeable perf regression we can switch to that option.
+    // Overriding the FORMAT_TEXT_COMMAND and PASTE_COMMAND commands is not an option I considered because
+    // there might be other forms of mutation that we might not be considering. For example:
+    // browser extensions or Payload/Lexical plugins that have their own commands.
+    return editor.registerNodeTransform(TextNode, (textNode) => {
+      disabledFormats.forEach((disabledFormat) => {
+        if (textNode.hasFormat(disabledFormat)) {
+          textNode.toggleFormat(disabledFormat)
+        }
+      })
+    })
+  }, [editor, features])
+
+  return null
+}
+
+function getDisabledFormats(enabledFormats: TextFormatType[]): TextFormatType[] {
+  // not sure why Lexical added highlight as TextNode format.
+  // see https://github.com/facebook/lexical/pull/3583
+  // We are going to implement it in other way to support multiple colors
+  delete TEXT_TYPE_TO_FORMAT.highlight
```

Comment:
@GermanJablo I do see why, from the use case described in the PR. I had implemented a custom feature relying on this format, which has broken because of this change. Is there a reason `highlight` must be deleted? After all, if you are taking a different approach, this is creating unexpected behaviour to a seemingly unrelated feature.

For my use case, it is sufficient to update the `enabledFormats` type and just pass it from my feature to enable `highlight`. Let me know if you think this makes sense, then I'll open a PR!

Reply:
The purpose of this PR is to allow users to enable or disable features of their choice. If we allow that option when we add our custom highlight, there would be two ways of doing the same thing that would conflict.

I'm sorry about this inconvenience. We're working on making it easier to implement features without the need to extend nodes.

Reply:
Sorry for the delay in responding.

I've been thinking about what you said, and discussed it internally. I think you're right. There should be a way to enable the default highlighting that comes with Lexical.

However, we don't want to have a HighlightFeature and a TextColorFeature that overlap in responsibilities. When we make a TextColorFeature, we'll put in an option for certain colors to have the highlight property in Lexical.

In the meantime, you may have to use a patch-package to comment out that line. Thanks for the patience and understanding!

Reply:
That's far from ideal, but up to you. Thanks for your reply! Note that I opened a PR (https://github.com/payloadcms/payload/pull/10415), feel free to handle that accordingly.